src/debug.c

```diff
@@ -0,0 +1,244 @@
+   (VoidFunc)IS_DENSE_LIST,
+   (VoidFunc)IS_DENSE_PLIST,
+   (VoidFunc)IS_FFE,
+   (VoidFunc)IS_HOMOG_LIST,
+   (VoidFunc)IS_INTOBJ,
+   (VoidFunc)IS_LIST,
+   (VoidFunc)IS_LVARS_OR_HVARS,
+   (VoidFunc)IS_MODULE_BUILTIN,
+   (VoidFunc)IS_MODULE_DYNAMIC,
+   (VoidFunc)IS_MODULE_STATIC,
+   (VoidFunc)IS_MUTABLE_PLIST,
+   (VoidFunc)IS_NONNEG_INTOBJ,
+   (VoidFunc)IS_PLIST,
+   (VoidFunc)IS_PLIST_OR_POSOBJ,
+   (VoidFunc)IS_POS_INTOBJ,
+   (VoidFunc)IS_POSS_LIST,
+   (VoidFunc)IS_PPERM,
+   (VoidFunc)IS_PREC,
+   (VoidFunc)IS_PREC_OR_COMOBJ,
+   (VoidFunc)IS_RANGE,
+   (VoidFunc)IS_SMALL_LIST,
+   (VoidFunc)IS_SSORT_LIST,
+   (VoidFunc)IS_STRING,
+   (VoidFunc)IS_STRING_REP,
+   (VoidFunc)IS_TABLE_LIST,
+   (VoidFunc)IS_TRANS,
+   (VoidFunc)ISB_LIST,
+#ifdef HPCGAP
+   (VoidFunc)LCKS_FUNC,
+#endif
+   (VoidFunc)LEN_BLIST,
+   (VoidFunc)LEN_LIST,
+   (VoidFunc)LEN_PLIST,
+   (VoidFunc)LEN_PREC,
+   (VoidFunc)LENGTH,
+   (VoidFunc)MakeHighVars,
+   (VoidFunc)MakeImmString,
+   (VoidFunc)MakeString,
+   (VoidFunc)MarkBag,
+   (VoidFunc)NAME_FUNC,
+   (VoidFunc)NAMS_FUNC,
+   (VoidFunc)NARG_FUNC,
+   (VoidFunc)NEW_PLIST,
+   (VoidFunc)NEW_RANGE_NSORT,
+   (VoidFunc)NEW_RANGE_SSORT,
+   (VoidFunc)NEW_TRANS,
+   (VoidFunc)NEW_TRANS2,
+   (VoidFunc)NEW_TRANS4,
+   (VoidFunc)NewWord,
+   (VoidFunc)NLOC_FUNC,
+   (VoidFunc)NUM_RAT,
+   (VoidFunc)NUMBER_BLOCKS_BLIST,
+   (VoidFunc)PLAIN_LIST,
+   (VoidFunc)PLEN_SIZE_BLIST,
+   (VoidFunc)PopPlist,
+   (VoidFunc)POS_LIST,
+   (VoidFunc)PROF_FUNC,
+   (VoidFunc)PTR_BAG,
+   (VoidFunc)PushPlist,
+   (VoidFunc)RANK_PPERM,
+   (VoidFunc)RANK_TRANS,
+   (VoidFunc)RegisterStatWithHook,
+   (VoidFunc)RESIZE_WORD,
+   (VoidFunc)SET_BODY_FUNC,
+   (VoidFunc)SET_CHAR_VALUE,
+   (VoidFunc)SET_DEN_RAT,
+   (VoidFunc)SET_ELM_PLIST,
+   (VoidFunc)SET_ELM_PREC,
+   (VoidFunc)SET_ELM_STRING,
+   (VoidFunc)SET_ENVI_FUNC,
+   (VoidFunc)SET_FEXS_FUNC,
+   (VoidFunc)SET_HDLR_FUNC,
+   (VoidFunc)SET_INC_RANGE,
+#ifdef HPCGAP
+   (VoidFunc)SET_LCKS_FUNC,
+#endif
+   (VoidFunc)SET_LEN_BLIST,
+   (VoidFunc)SET_LEN_PLIST,
+   (VoidFunc)SET_LEN_PREC,
+   (VoidFunc)SET_LEN_RANGE,
+   (VoidFunc)SET_LEN_STRING,
+   (VoidFunc)SET_LOW_RANGE,
+   (VoidFunc)SET_NAMS_FUNC,
+   (VoidFunc)SET_NARG_FUNC,
+   (VoidFunc)SET_NLOC_FUNC,
+   (VoidFunc)SET_NUM_RAT,
+   (VoidFunc)SET_OBJ_FLAG,
+   (VoidFunc)SET_PROF_FUNC,
+   (VoidFunc)SET_PTR_BAG,
+   (VoidFunc)SET_RNAM_PREC,
+   (VoidFunc)SET_VAL_MACFLOAT,
+   (VoidFunc)SetBrkCallTo,
+   (VoidFunc)SHRINK_PLIST,
+   (VoidFunc)SHRINK_STRING,
+   (VoidFunc)SIZE_BAG,
+   (VoidFunc)SIZE_BAG_CONTENTS,
+   (VoidFunc)SIZE_OBJ,
+   (VoidFunc)SIZE_PLEN_BLIST,
+   (VoidFunc)SIZEBAG_STRINGLEN,
+   (VoidFunc)SwitchToNewLvars,
+   (VoidFunc)SwitchToOldLVars,
+   (VoidFunc)SwitchToOldLVarsAndFree,
+   (VoidFunc)TEST_OBJ_FLAG,
+   (VoidFunc)TNAM_OBJ,
+   (VoidFunc)TNUM_BAG,
+   (VoidFunc)TNUM_OBJ,
+   (VoidFunc)UNB_LIST,
+   (VoidFunc)VAL_MACFLOAT,
+   (VoidFunc)VisitStatIfHooked,
+};
```

Comment:
Bah, I really don't like this, but I agree it fixes a problem and I don't have a better answer.

GCC has a special option to fix this, `-fkeep-inline-functions`, but it seems clang doesn't support that, or anything similar.

Reply:
In addition to not being portable (and there are more compilers than `gcc` and `clang` too, and hopefully more again in the future), the other problem with `-fkeep-inline-functions` is that it actually leads to *tons* of copies of these functions being emitted... Anyway, the above is perhaps not "nice", but simply works and is portable. We could even write a script to update the array for us.